### PR TITLE
feat(atomic): Add insight result attach-to-case component

### PR DESCRIPTION
**Pull request by jfallaire**
Introduce the AtomicInsightResultAttachToCase component, which emits two distinct events:

* `atomic/insight/attachToCase/attach`
* `atomic/insight/attachToCase/detach`

By listening to these events, you will gain access to the attachToCase controller and the associated result.

The component can also be used as a visual indicator when results are attached (readonly prop)

### CODE REVIEW

File: packages/atomic/src/pages/examples/insights.html [hunk 1]
```diff
@@ -441,6 +457,22 @@
       document.addEventListener('atomicInsightResultActionClicked', (event) => {
         console.log('Result action clicked', event.detail);
       });
+
+      document.addEventListener('atomic/insight/attachToCase/attach', (event) => {
+        console.log('attach', event.detail);
```
Comment (SimonMilord) [inline packages/atomic/src/pages/examples/insights.html:462]:
```suggestion
```

**Review by SimonMilord — APPROVED** [review]

File: packages/atomic/src/components/insight/atomic-insight-result-attach-to-case/atomic-insight-result-attach-to-case.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
+import {AttachToCase, buildAttachToCase, Result} from '@coveo/headless/insight';
```
Comment (fbeaudoincoveo) [inline packages/atomic/src/components/insight/atomic-insight-result-attach-to-case/atomic-insight-result-attach-to-case.tsx:1]: Did you consider splitting this into two components (e.g., `atomic-insight-result-attach-to-case-action` and `atomic-insight-result-attach-to-case-indicator`)?

It feels like a somewhat odd design to me to have the same component used in two different ways based on the value of a prop.

File: packages/atomic/src/components/insight/atomic-insight-result-action-bar/atomic-insight-result-action-bar.pcss [hunk 3]
```diff
@@ -16,4 +16,18 @@ atomic-insight-result-action-bar {
...
+      &:hover {
+        @apply border-r;
+      }
+    }
+  }
+  > atomic-insight-result-attach-to-case:not(:first-child) {
+    button {
+      @apply rounded-bl-none rounded-tl-none;
+    }
+  }
```
[the remark this answers is not in the document]
Comment (jfallaire) [reply packages/atomic/src/components/insight/atomic-insight-result-action-bar/atomic-insight-result-action-bar.pcss:32]: Yes make sense, let's do that instead.

File: packages/atomic/src/pages/examples/insights.html [hunk 4]
```diff
@@ -441,6 +457,22 @@
...
+        const {controller, result} = event.detail;
+
+        new Promise(resolve => setTimeout(() => resolve(), 1000)).then(() => {
```
[the remark this answers is not in the document]
Comment (jfallaire) [reply packages/atomic/src/pages/examples/insights.html:465]: Yes my bad, forgot to remove this. it was for a small test.

File: packages/atomic/src/components/insight/atomic-insight-result-attach-to-case/atomic-insight-result-attach-to-case.tsx [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+import {IconButton} from '../../common/iconButton';
+import {ResultContext} from '../../search/result-template-components/result-template-decorators';
+import {InsightBindings} from '../atomic-insight-interface/atomic-insight-interface';
+
+export interface InsightResultAttachToCaseEvent {
+  controller: AttachToCase;
+  result: Result;
+}
+/**
+ * @internal
```
[the remark this answers is not in the document]
Comment (jfallaire) [reply packages/atomic/src/components/insight/atomic-insight-result-attach-to-case/atomic-insight-result-attach-to-case.tsx:27]: Thanks! For the short term, I think we should keep this internal. The goal is to establish a more effective hook for the ongoing ServiceNow integration.

File: packages/atomic/src/components/insight/atomic-insight-result-attach-to-case/atomic-insight-result-attach-to-case.tsx (hunk 2 above)
Comment (jfallaire) [reply packages/atomic/src/components/insight/atomic-insight-result-attach-to-case/atomic-insight-result-attach-to-case.tsx:1]: Good point, I agree it would make more sense. @mikegron what do you think?

File: packages/atomic/src/components/insight/atomic-insight-result-attach-to-case/atomic-insight-result-attach-to-case.tsx [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+      this.bindings.engine.state.insightCaseContext?.caseId || '';
+    this.attachToCase = buildAttachToCase(this.bindings.engine, {
+      options: {
+        result: this.result,
+        caseId: caseId,
+      },
+    });
+  }
+
+  private onClick() {
```
Comment (mmitiche) [inline packages/atomic/src/components/insight/atomic-insight-result-attach-to-case/atomic-insight-result-attach-to-case.tsx:89]: Why are we not calling the the methods attach and detach of the `attachToCase` controller on click?

File: packages/atomic/src/components/insight/atomic-insight-result-attach-to-case/atomic-insight-result-attach-to-case.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (fbeaudoincoveo) [reply packages/atomic/src/components/insight/atomic-insight-result-attach-to-case/atomic-insight-result-attach-to-case.tsx:27]: That being said, addings e2e tests would probably be a good idea to make sure we don't accidentally introduce breaking behavior changes / regressions in the future.

File: packages/atomic/src/components/insight/atomic-insight-result-attach-to-case/atomic-insight-result-attach-to-case.tsx (hunk 2 above)
Comment (mikegron) [reply packages/atomic/src/components/insight/atomic-insight-result-attach-to-case/atomic-insight-result-attach-to-case.tsx:1]: J'enleverais la prop au complet, ca ne fait "rien" anyway si personne se hook sur l'event donc ont a pas besoin de ca (j'imagine pour le builder?)

Comment (alexprudhomme) [reply packages/atomic/src/components/insight/atomic-insight-result-attach-to-case/atomic-insight-result-attach-to-case.tsx:1]: > It feels like a somewhat odd design to me to have the same component used in two different ways based on the value of a prop.

I agree with this. I think it should be two components.

File: packages/atomic/src/pages/examples/insights.html [hunk 7]
```diff
@@ -441,6 +457,18 @@
...
+        console.log('attach', event.detail);
+        const {controller, result} = event.detail;
+        controller.attach();
+      });
+
+      document.addEventListener('atomic/insight/attachToCase/detach', (event) => {
+        console.log('detach', event.detail);
+        const {controller, result} = event.detail;
+        controller.detach();
+      });
```
Comment (alexprudhomme) [inline packages/atomic/src/pages/examples/insights.html:471]: Can you explain why we need those javascript event listeners ? Couldn't this be done via stencil events and without those events in the implementation file ? 

https://stenciljs.com/docs/events

File: packages/atomic/src/pages/examples/insights.html [hunk 8]
```diff
@@ -123,13 +127,17 @@
...
+                  <atomic-insight-result-attach-to-case></atomic-insight-result-attach-to-case>
                   <atomic-insight-result-action tooltip="quickview" action="quickview"></atomic-insight-result-action>
                 </atomic-insight-result-action-bar>
                 <atomic-result-section-badges>
                   <atomic-field-condition must-match-sourcetype="Salesforce">
                     <atomic-result-badge label="Salesforce" class="salesforce-badge"></atomic-result-badge>
                   </atomic-field-condition>
                 </atomic-result-section-badges>
+                <atomic-result-section-actions
+                  ><atomic-insight-result-attach-to-case read-only="true"></atomic-insight-result-attach-to-case
```
Comment (alexprudhomme) [inline packages/atomic/src/pages/examples/insights.html:139]: I had this UI issue when there was no badge. Not that big of a problem imo

https://github.com/user-attachments/assets/1676bb9b-f7f8-4d39-b15a-2cc2429b7551

File: packages/atomic/src/components/insight/atomic-insight-result-attach-to-case/atomic-insight-result-attach-to-case.tsx (hunk 2 above)
Comment (jfallaire) [reply packages/atomic/src/components/insight/atomic-insight-result-attach-to-case/atomic-insight-result-attach-to-case.tsx:1]: I will make two components. Thanks for your feedback

File: packages/atomic/src/components/insight/atomic-insight-result-attach-to-case/atomic-insight-result-attach-to-case.tsx (hunk 6 above)
Comment (jfallaire) [reply packages/atomic/src/components/insight/atomic-insight-result-attach-to-case/atomic-insight-result-attach-to-case.tsx:89]: The goal is to make this generic enough to support real attach/detach actions in ServiceNow, or any other ecosystem using the hosted insight approach. By leveraging the exposed event listeners (attach/detach) and passing the controller, the implementer will be responsible for calling the attach/detach methods.

File: packages/atomic/src/pages/examples/insights.html (hunk 8 above)
Comment (mikegron) [reply packages/atomic/src/pages/examples/insights.html:139]: Looks like a pretty big issue 😆

**Review by dguerinCoveo — APPROVED** [review]

File: packages/atomic/src/components/insight/atomic-insight-result-attach-to-case-action/atomic-insight-result-attach-to-case-action.tsx [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+        caseId: caseId,
+      },
+    });
+  }
+
+  private onClick() {
+    if (this.attachToCase.isAttached()) {
+      this.detach.emit({controller: this.attachToCase, result: this.result});
+    } else {
+      this.attach.emit({controller: this.attachToCase, result: this.result});
```
Comment (mmitiche) [inline packages/atomic/src/components/insight/atomic-insight-result-attach-to-case-action/atomic-insight-result-attach-to-case-action.tsx:76]: same remark here, instead of passing `this.attachToCase`, pass `() => {this.attachToCase.attach()}`

**Review by mmitiche — APPROVED** [review]
Small suggestion about the payload of the emitted event, but looks good to me.

**Review by fbeaudoincoveo — APPROVED** [review]
